src/app/WriteClient.cpp

```diff
@@ -303,13 +305,13 @@ CHIP_ERROR WriteClient::OnMessageReceived(Messaging::ExchangeContext * apExchang
         ExitNow();
     }
 
+    err = ProcessWriteResponseMessage(std::move(aPayload));
+
     // Close the current exchange after receiving the response since the response message marks the
     // end of conversation represented by the exchange. We should create an new exchange for a new
     // conversation defined in Interaction Model protocol.
     ClearExistingExchangeContext();
```

Comment:
This aborts the exchange, so never sends acks for a write response.  Is there a reason this method looks totally different from command and read?  Especially since it's clearly a copy/paste of the command code at some point (see the "Invoke Command Response" comment a few lines above...)

Reply:
it should call close instead of abort when clear exchange context.

Reply:
> so never sends acks for a write response

I was wrong here.  Abort does send acks; it just does not send any more retransmits.

Arguably, if we got the app-level reply here the other side got our request even if we have not gotten an ack for it, so aborting here is actually not entirely unreasonable.